includes/specials/SpecialListgrouprights.php

```diff
@@ -60,7 +60,10 @@ public function execute( $par ) {
 				'</tr>'
 		);
 
-		$allGroups = User::getAllGroups();
+		$allGroups = array_unique( array_merge(
+			User::getAllGroups(),
```

Comment:
You are right, I had the same thought - that is exactly the reason why in the new internal service I renamed it to getExplicitGroups. I didn't want to change the API of the User class however to avoid problems and the need to refactor all places that use it to create even more havoc, so this name is simply in the form it existed in the past.

What I can do here however is to use the new service instead of the User class, which has much better naming and decrease the number of dependencies on the old badly named User API